tapfreighter/chain_porter.go

```diff
@@ -768,16 +768,18 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 			return nil, fmt.Errorf("unable to cast parcel to " +
 				"address parcel")
 		}
-		fundSendRes, err := p.cfg.AssetWallet.FundAddressSend(
-			ctx, addrParcel.destAddrs...,
-		)
+		fundSendRes, outputIdxToAddr, err :=
+			p.cfg.AssetWallet.FundAddressSend(
+				ctx, addrParcel.destAddrs...,
+			)
 		if err != nil {
 			return nil, fmt.Errorf("unable to fund address send: "+
 				"%w", err)
 		}
 
 		currentPkg.VirtualPacket = fundSendRes.VPacket
 		currentPkg.InputCommitments = fundSendRes.InputCommitments
+		currentPkg.OutputIdxToAddr = outputIdxToAddr
```

Comment:
This will work for a send where only the chain porter is involved throughout the process (e.g. from an `AddressParcel`). But it will fail once we do actual PSBT funding (meaning there is no in-memory state in the chain porter between the `SendStateVirtualCommitmentSelect` and the `SendStateVirtualSign` steps, because all state is purely kept in the vPSBT by the caller between the different RPC calls that trigger these steps).

So unless we want to refactor the whole chain porter state machine to become stateful across RPC calls (and with that also across restarts), the only "place" that this information can be stored between calls is the vPSBT itself.

The reason why no integration test currently fails is because we still manually export and import the proof in those tests instead of using a courier (we should probably change that soon as well).

Reply:
Discussed this offline and it sounds like we're okay with this for now, since as soon as things are somewhat interactive, the proof transfer is expected to happen outside of the porter/courier as well.